lib/game_rules.ex

```diff
@@ -1,6 +1,12 @@
 defmodule GameRules do
-  def over?(board) do
+  def player_lost?(board) do
     Board.all_tiles(board)
     |> Enum.any?(fn tile -> Tile.exploded?(tile) end)
   end
+
+  def player_won?(board) do
+    Board.all_tiles(board)
+    |> Enum.filter(fn tile -> Tile.is_empty?(tile) end)
```

Comment:
Have you heard the term _arity_ before? I'm gonna explain it anyways, but skip ahead if you have. :smile: 

Arity is a ten-dollar functional programming word for "the number of arguments a function takes." (It comes from unary, binary, ternary, etc.) Your anonymous function `fn tile -> Tile.is_empty?(tile) end` is "unary", "1-ary" or "has arity of 1."

Now take a look at the body of that anonymous function: `Tile.is_empty?(tile)`. What is the arity of the `is_empty` function? It's also 1-ary, right? More important than just taking one argument, it takes _the same one argument_ as the anonymous function that's wrapping it. 

That means the anonymous function isn't doing much besides providing familiar syntax to pass to `Enum.filter`: it takes in an arg and passes it right along to another function. What if you could just pass the `is_empty` function itself?

In fact you can in Elixir (and other functional languages, and many OO languages and other paradigms, too). You'll hear this called "passing functions as a parameter," "passing functions around", and ["first class functions"](https://docs.microsoft.com/en-us/dotnet/fsharp/introduction-to-functional-programming/first-class-functions). Part of the power of functional languages is that you can treat functions as values: store them in data structures like lists and maps, assign them to variables, pass them as parameters to other ("higher-order") functions.

In languages that use parens to indicate evaluation (Python, Javascript), you can usually just pass around the function name. (e.g. `roflcopter()` is a function call, while `roflcopter` is a value). Ruby and Elixir are different, since parens are optional for mostly-I-guess-aesthetic reasons, which means `Lolwut.roflcopter()` and `Lolwut.roflcopter` are both function calls. That means in both languages, you have to use special syntax to pass these functions as values. There are a zillion ways to do this in Ruby, but you've probably seen something like `[1, 2, 3].reduce(:+)` (passes the `+` function as a symbol), or `["a", "b", "c"].map(&:upcase)` (which is actually a clever way of passing [a block](https://skorks.com/2013/04/ruby-ampersand-parameter-demystified/)). 

To pass a function in Elixir, you use syntax like `&Module.function/arity`. That is: 1) an IMO very ugly ampersand, 2) the function's module and name, and 3) the arity of the function, e.g. `&Lolwut.roflcopter/0`, `&Tile.is_empty?/1`. (Since Elixir lets you define multiple functions inside a module that have the same name but different arities, you have to specify the arity so Elixir can figure out which function you're passing).

So this is all to say I think you can try passing the `Tile.is_empty` function (err, `&Tile.is_empty?/1`) as a _value_ here (and any time you find yourself with an anonymous function that hands its arguments off directly to another function without doing anything else).

Reply:
lol, I see you used this syntax in your own PR description and a bunch of places in your code, so I hope this isn't pedantic mentorsplaining! 🙃

Reply:
I love this idea, too! :) And thank you for the great explanation. I also love the term "mentorsplaining," but I don't think this is a case of it!